spikeextractors/extractors/neoextractors/neobaseextractor.py

```diff
@@ -80,6 +92,48 @@ def get_channel_ids(self):
         return chan_ids
 
 
-# Not done yet
-class NeoBaseSortingExtractor(SortingExtractor):
-    NeoRawIOClass = None
+class NeoBaseSortingExtractor(SortingExtractor, _NeoBaseExtractor):
+    def __init__(self, **kargs):
+        _NeoBaseExtractor.__init__(self, **kargs)
+        
+        # the sampling frequency is quite tricky because in neo
+        # spike are handle in s or ms
+        # internally many format do have have the spike time stamps
+        # at the same speed as the signal but at a higher clocks speed.
+        # here in spikeinterface we need spike index to be at the same speed 
+        # that signal it do not make sens to have spikes at 50kHz sample
+        # when the sig is 10kHz.
+        # neo handle this but not spikeextractors
+        
+        self._handle_sampling_frequency()
+        #self._sampling_frequency = None
+    
+    def _handle_sampling_frequency(self):
+        # must handle :
+        #   _sampling_frequency
+        #   _time_stamp0
+        raise(NotImplementedError)
+    
+    def get_unit_ids(self):
+        # should be this
+        unit_ids = self.neo_reader.header['unit_channels']['id']
+        # in neo unit_ids are string so here we take index
+        unit_ids = np.arange(unit_ids.size, dtype='int64')
+        return unit_ids
+    
+    def get_unit_spike_train(self, unit_id, start_frame=None, end_frame=None):
+        neo_unit_id = self.neo_reader.header['unit_channels']['id'][unit_id]
+        
+        # neo handle limts in times (s, ms) but spikeextractors in frames.
+        # but spike can have diffrents sampling rate than signals
+        # so conversion from signals frames to times is
+        # format dependent
+        assert start_frame is None , 'Do not handle slice of spikes'
```

Comment:
Will this SortingExtractor not work when giving start and end frames?

Reply:
Yes. For the moment.
It is too complicated. I need a call with you and Alessio on this point.
Next week maybe.

Reply:
Yes.
This is a better option.
An the neo lazy read of spike is very slow because you can seek in te file.

Reply:
@samuelgarcia can you change this? then it's ready to merge for me